server/template.go

```diff
@@ -78,3 +82,87 @@ input:focus::-moz-placeholder{color:transparent}
 input:focus:-moz-placeholder{color:transparent}
 input:focus:-ms-input-placeholder{color:transparent}`
 )
+
+// indexJS is the javascript for the main index page.
+func indexJS(defaultProvider string) string {
+	var b bytes.Buffer
+
+	b.WriteString(`function sInit() {
+    "use strict";
+    var searchForm = document.forms.search;
```

Comment:
Re: My last comment, you actually don't even need to reference `document.forms`; when you give an element an ID, it becomes available on the `window` (global) object.